zerver/models.py

```diff
@@ -1561,6 +1561,15 @@ def active_user_ids(realm_id: int) -> List[int]:
     ).values_list('id', flat=True)
     return list(query)
 
+@cache_with_key(active_non_guest_user_ids_cache_key, timeout=3600*24*7)
```

Comment:
This is the first time I've dealt with caching, I've to add this because backend tests failed where there were  ~110 quires instead of expected ~75.

Reply:
Yep, we definitely want a cache here.  That said ... we should look more closely; it sorta sounds like we might be at risk of fetching this in a loop from memcached, which probably is also wrong.

Reply:
>  it sorta sounds like we might be at risk of fetching this in a loop from Memcached, which probably is also wrong.

If this comes as a conclusion of the fact "~110 quires instead of ~75", then I've tried tests without caching for `active_user_ids` on `master` and saw similar behavior:
```
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.001', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.001', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.001', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}
{'time': '0.000', 'sql': 'SELECT "zerver_userprofile"."id" FROM "zerver_userprofile" WHERE ("zerver_userprofile"."is_active" = true AND "zerver_userprofile"."realm_id" = 1)'}

expected length: 21
actual length: 40
```
Which is similar to what observed without caching for `active_non_guest_user_ids`.